File: 5-heap/Main.java
import java.util.*;

class Main {
  public static void main(String[] args) throws Exception {
    List<Integer> l = new ArrayList<>();
    l.add(4);
    l.add(7);
    l.add(9);
    l.add(1);
    l.add(5);
    l.add(12);
    l.add(7);
    l.add(2);
    MaxHeap m = new MaxHeap(l);
    System.out.println(m.getMax());
    var h = m.getHeap();
    for (int i : h) {
      System.out.printf("%d ", i);
    }
    System.out.println();
  }
}

File: 5-heap/main.cpp
#include <iostream>
#include <vector>

#include "MaxHeap.hpp"

using namespace std;

int main() {
  MaxHeap m({4, 7, 9, 1, 5, 12, 7, 2});
  // m.insert(4);
  // m.insert(7);
  // m.insert(9);
  // m.insert(1);
  // m.insert(5);
  // m.insert(12);
  // m.insert(7);
  // m.insert(2);
  // m.remove();
  auto v = m.getHeap();
  for (auto i : v) {
    cout << i << " ";
  }
  cout << endl;
  return 0;
}